services/winget/winget-version.service.js

```diff
@@ -0,0 +1,97 @@
+import Joi from 'joi'
+import gql from 'graphql-tag'
+import { latest, renderVersionBadge } from '../version.js'
+import { InvalidParameter, pathParam } from '../index.js'
+import { GithubAuthV4Service } from '../github/github-auth-service.js'
+import { transformErrors } from '../github/github-helpers.js'
+
+const schema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      object: Joi.object({
+        entries: Joi.array().items(
+          Joi.object({
+            type: Joi.string().required(),
+            name: Joi.string().required(),
+          }),
+        ),
+      })
+        .allow(null)
+        .required(),
+    }).required(),
+  }).required(),
+}).required()
+
+export default class WingetVersion extends GithubAuthV4Service {
+  static category = 'version'
+
+  static route = {
+    base: 'winget/v',
+    pattern: ':name',
+  }
+
+  static openApi = {
+    '/winget/v/{name}': {
+      get: {
+        summary: 'WinGet Package Version',
+        description: 'WinGet Community Repository',
+        parameters: [
+          pathParam({
+            name: 'name',
+            example: 'Microsoft.WSL',
+          }),
+        ],
+      },
+    },
+  }
+
+  static defaultBadgeData = {
+    label: 'winget',
+  }
+
+  async fetch({ name }) {
+    const nameFirstLower = name[0].toLowerCase()
+    const nameSlashed = name.replaceAll('.', '/')
+    const path = `manifests/${nameFirstLower}/${nameSlashed}`
+    const expression = `HEAD:${path}`
+    return this._requestGraphql({
+      query: gql`
+        query RepoFiles($expression: String!) {
+          repository(owner: "microsoft", name: "winget-pkgs") {
+            object(expression: $expression) {
+              ... on Tree {
+                entries {
+                  type
+                  name
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: { expression },
+      schema,
+      transformErrors,
+    })
+  }
+
+  async handle({ name }) {
+    try {
+      const json = await this.fetch({ name })
+      if (json.data.repository.object === null) {
+        throw new InvalidParameter({
+          prettyMessage: 'package not found',
+        })
+      }
+      const entries = json.data.repository.object.entries
+      const directories = entries.filter(file => file.type === 'tree')
+      const versions = directories.map(file => file.name)
+      const version = latest(versions)
+
+      return renderVersionBadge({ version })
+    } catch (e) {
```

Comment:
Are we expecting to catch a particular exception here?
If not, can we ditch the try/catch. Sentry will handle capturing the details of any unexpected runtime error for us in production.

Reply:
sorry I forgot to remove debug code.

Reply:
removed in d9963b5f5af7c66f11bbdfff1d44a2ae444fce22